Sources/Client.swift

```diff
@@ -0,0 +1,334 @@
+//
+//  Copyright © 2019 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+public protocol ClientDelegate: class {
+
+    func client(_ client: Client, didFailToConnect url: WCURL)
+    func client(_ client: Client, didConnect session: Session)
+    func client(_ client: Client, didDisconnect session: Session)
+
+}
+
+public class Client: WalletConnect {
+
+    public typealias RequestResponse = (Response) -> Void
+
+    private(set) weak var delegate: ClientDelegate!
+    private let dAppInfo: Session.DAppInfo
+    private var responses: Responses
+
+    enum ClientError: Error {
+        case missingWalletInfoInSession
+        case missingRequestID
+        case sessionNotFound
+    }
+
+    public init(delegate: ClientDelegate, dAppInfo: Session.DAppInfo) {
+        self.delegate = delegate
+        self.dAppInfo = dAppInfo
+        responses = Responses(queue: DispatchQueue(label: "org.walletconnect.swift.client.pending"))
+        super.init()
+    }
+
+    /// Send request to wallet.
+    ///
+    /// - Parameters:
+    ///   - request: Request object.
+    ///   - completion: RequestResponse completion.
+    /// - Throws: Client error.
+    public func send(_ request: Request, completion: RequestResponse?) throws {
+        guard let session = communicator.session(by: request.url) else {
+            throw ClientError.sessionNotFound
+        }
+        guard let walletInfo = session.walletInfo else {
+            throw ClientError.missingWalletInfoInSession
+        }
+        guard let requestID = request.payload.id else {
+            throw ClientError.missingRequestID
+        }
+        if let completion = completion {
+            responses.add(requestID: requestID, response: completion)
+        }
+        communicator.send(request, topic: walletInfo.peerId)
+    }
+
+    /// Send response to wallet.
+    ///
+    /// - Parameter response: Response object.
+    /// - Throws: Client error.
+    public func send(_ response: Response) throws {
+        guard let session = communicator.session(by: response.url) else {
+            throw ClientError.sessionNotFound
+        }
+        guard let walletInfo = session.walletInfo else {
+            throw ClientError.missingWalletInfoInSession
+        }
+        communicator.send(response, topic: walletInfo.peerId)
+    }
+
+    /// https://docs.walletconnect.org/json-rpc/ethereum#personal_sign
```

Comment:
I think the URLs should be on the 2nd or 3rd line, as the description of the method is better. This is how it is done in the Swift Markup:

```
/// - SeeAlso: https://docs.walletconnect.org/json-rpc/ethereum#personal_sign
```

All the other method docs below ⬇️  - the same comment